vmm/sandbox/src/lib.rs

```diff
@@ -44,22 +45,21 @@ pub const NAMESPACE_MNT: &str = "mount";
 pub const NAMESPACE_CGROUP: &str = "cgroup";
 
 async fn load_config<T: DeserializeOwned>(
+    args: &Args,
     default_config_path: &str,
 ) -> anyhow::Result<(Config<T>, String)> {
-    let os_args: Vec<_> = std::env::args_os().collect();
     let mut config_path = default_config_path.to_string();
     let mut dir_path = String::new();
-    for i in 0..os_args.len() {
-        if os_args[i].to_str().unwrap() == "--config" {
-            config_path = os_args[i + 1].to_str().unwrap().to_string()
-        }
-        if os_args[i].to_str().unwrap() == "--dir" {
-            dir_path = os_args[i + 1].to_str().unwrap().to_string();
-            if !std::path::Path::new(&dir_path).exists() {
-                tokio::fs::create_dir_all(&dir_path).await.unwrap();
-            }
+    if let Some(c) = &args.config {
+        config_path = c.to_string();
+    }
+    if let Some(d) = &args.dir {
+        dir_path = d.to_string();
+        if !std::path::Path::new(&dir_path).exists() {
+            tokio::fs::create_dir_all(&dir_path).await.unwrap();
```

Comment:
```suggestion
            tokio::fs::create_dir_all(&dir_path).await.with_context(|| format!("Failed to mkdir for {}", dir_path))?;
```
Should be formatted.